sm_value_store/src/VerboseValueStore.cpp

```diff
@@ -52,8 +52,14 @@ std::string getValue(const ValueHandle<std::string> & v) {
 template <typename T, typename O>
 T VerboseValueStore::logValue(const char * func, const std::string & path, T && v, O o) const {
   std::string extra = "";
-  if(!std::is_same<bool, O>::value && o){
-    if(!vs_->hasKey(path)){
+  //prevent unused warning
+  static_cast<void>(o);
+  if constexpr(!std::is_same_v<std::false_type, O>) {
+    //TODO (huberya): we somehow want to switch between optional (and others) for DEFAULT vs INITIAL (@Hannes)
+    /*if constexpr(std::is_same_v<std::decay_t<decltype(std::declval<T>.get())>, std::decay_t<O>>) {
+      extra = "INITIAL:";
+    }
+    else */if(o && !vs_->hasKey(path)) {
```

Comment:
Don't we even want a overall different log message? How much can we really reuse from this function? For instance for adding the initial is always provided and after adding the key should always be there. 
Would it make sense to simply have another `logAddValue` function for the adding? (of course also using `log_` and `getValue`.